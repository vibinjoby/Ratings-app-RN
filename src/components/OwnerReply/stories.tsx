import React from 'react'
import { storiesOf } from '@storybook/react-native'
import { StyleSheet, View } from 'react-native'

import OwnerReply from './component'

storiesOf('Components', module).add('OwnerReply', () => (
  <View style={styles.container}>
    <OwnerReply owner_reply="Hey sorry that we couldn't meet your expectations" />
  </View>
))

const styles = StyleSheet.create({
  container: {
    padding: 20,
  },
})
