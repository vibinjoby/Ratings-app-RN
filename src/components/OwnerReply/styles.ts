import { StyleSheet } from 'react-native'

import Colors from '../../utilities/colors'
import Typography from '../../utilities/typography'

export default StyleSheet.create({
  container: {
    flex: 1,
  },
  ownerWrapper: {
    borderLeftWidth: 4,
    paddingLeft: 10,
    paddingTop: 5,
    borderLeftColor: Colors.borderColor,
  },
  ownerTitleWrapper: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  ownerPic: {
    width: 30,
    height: 30,
  },
  ownerTitle: {
    ...Typography.Body.bold,
    fontSize: 14,
    marginLeft: 10,
  },
})
