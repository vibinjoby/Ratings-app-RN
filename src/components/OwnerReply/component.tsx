import React from 'react'
import { Image, Text, View } from 'react-native'
import styles from './styles'

export interface OwnerReplyProps {
  /* eslint-disable camelcase */ owner_reply: string
}
/* eslint-disable camelcase */
const OwnerReply: React.FC<OwnerReplyProps> = ({ owner_reply }: OwnerReplyProps) => {
  return (
    <View style={styles.ownerWrapper}>
      <View style={styles.ownerTitleWrapper}>
        <Image source={require('../../assets/guest-pic.png')} style={styles.ownerPic} />
        <Text testID="ownerTitle" style={styles.ownerTitle}>
          Business Owner
        </Text>
      </View>
      {/* eslint-disable camelcase */}
      <Text testID="ownerReply">{owner_reply}</Text>
    </View>
  )
}

export default OwnerReply
