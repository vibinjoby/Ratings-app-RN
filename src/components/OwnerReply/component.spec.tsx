import 'react-native'
import React from 'react'
import { render } from '@testing-library/react-native'

import OwnerReply from './component'

describe('OwnerReply', () => {
  const MOCK_REPLY = "Hey sorry that we couldn't meet your expectations"

  it('should render', () => {
    const { toJSON } = render(<OwnerReply ownerReply={MOCK_REPLY} />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('should render reply', () => {
    const { getByTestId } = render(<OwnerReply ownerReply={MOCK_REPLY} />)
    expect(getByTestId('ownerTitle').props.children).toBe('Business Owner')
    expect(getByTestId('ownerReply').props.children).toBe(MOCK_REPLY)
  })
})
