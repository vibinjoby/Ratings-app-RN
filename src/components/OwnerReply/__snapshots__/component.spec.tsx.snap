// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OwnerReply should render 1`] = `
<View
  style={
    Object {
      "borderLeftColor": "#E2E2E2",
      "borderLeftWidth": 4,
      "paddingLeft": 10,
      "paddingTop": 5,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/guest-pic.png",
        }
      }
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#181725",
          "fontFamily": "Poppins-Bold",
          "fontSize": 14,
          "marginLeft": 10,
        }
      }
      testID="ownerTitle"
    >
      Business Owner
    </Text>
  </View>
  <Text
    testID="ownerReply"
  >
    Hey sorry that we couldn't meet your expectations
  </Text>
</View>
`;
